### PR TITLE
fix: accounts security

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- Fixes change the email button to be small
- Fixes set password button to be small
- Refactor the password field component to allow the option whether to automatically indicate password strength
- Fixes change email button should be hidden if the user doesn't have a password yet

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-571 #done
WT-572 #done
WT-573 #done
WT-574 #done

### CODE REVIEW

**Review by rebelchris — CHANGES_REQUESTED** [review]
I think we should refactor this.
It should never start validating without input or submit.

**Comment by sshanzel** [comment]
> I think we should refactor this.
It should never start validating without input or submit.

Totally agree with that.

**Comment by sshanzel** [comment]
@rebelchris I have added a few more fixes, specifically the missing tooltip and the styling of removing a provider. Let me know what you think 🚢

File: packages/webapp/components/layouts/AccountLayout/Security/index.tsx [hunk 1]
```diff
@@ -176,8 +204,13 @@ function AccountSecurityDefault({
             inputId="new_password"
             label="Password"
             name="password"
+            indicateStrengthAutomatically={false}
```
Comment (rebelchris) [inline packages/webapp/components/layouts/AccountLayout/Security/index.tsx:207]: Why do we choose to turn this off here?

File: packages/shared/src/components/fields/PasswordField.tsx [hunk 2]
```diff
@@ -45,20 +52,32 @@ export function PasswordField({
...
       {...props}
       autoComplete="off"
       type={useType}
-      valueChanged={(value) =>
-        setPasswordStrengthLevel(passwordStrength(value).id)
-      }
+      valueChanged={onChange}
       leftIcon={<LockIcon />}
-      hint={showStrength ? hint : props.hint}
+      hint={showStrength && indicateStrength ? hint : props.hint}
```
Comment (rebelchris) [inline packages/shared/src/components/fields/PasswordField.tsx:77]: I think this is becoming a little bit messy with mulitple states indicating if it should show or not?
Or maybe I'm misreading it.

But do we really need to have it so complex?
Why don't we simply always show the strength once inputted or validating?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Some small questions

File: packages/webapp/components/layouts/AccountLayout/Security/index.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/webapp/components/layouts/AccountLayout/Security/index.tsx:207]: The design team asked that we should not show any indicator until the user interacted with it. Well, I think this should not be needed anymore after we have decided to start checking the strength once the field has value.

File: packages/shared/src/components/fields/PasswordField.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/fields/PasswordField.tsx:77]: I agree, this was the previous work before we had that condition. Let me refactor this one.

File: packages/webapp/components/layouts/AccountLayout/Security/index.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/webapp/components/layouts/AccountLayout/Security/index.tsx]: Ah yes perfect, so we can remove this whole indicateauto stuff that would make it way more readable.

File: packages/shared/src/components/fields/PasswordField.tsx [hunk 3]
```diff
@@ -45,17 +46,20 @@ export function PasswordField({
     ? `Password need a minimum length of ${props.minLength}`
     : passwordStrengthStates[passwordStrengthLevel].label;
   const hint = !hasUserAction ? null : userActionHint;
-  const shouldShowStrength = showStrength && hasUserAction;
+  const shouldShowStrength = !!value && showStrength && hasUserAction;
```
Comment (rebelchris) [inline packages/shared/src/components/fields/PasswordField.tsx:49]: It should always validate on submit though, just making sure that happens (even when input is empty)

**Review by rebelchris — CHANGES_REQUESTED** [review]

File: packages/shared/src/components/fields/PasswordField.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/components/fields/PasswordField.tsx:49]: I think when it is empty, the hint should be different, that it is required/missing.

File: packages/shared/src/components/fields/PasswordField.tsx [hunk 4]
```diff
@@ -45,17 +46,20 @@ export function PasswordField({
...
     <TextField
       {...props}
       autoComplete="off"
       type={useType}
-      valueChanged={(value) =>
-        setPasswordStrengthLevel(passwordStrength(value).id)
-      }
+      valueChanged={onChange}
       leftIcon={<LockIcon />}
       hint={showStrength ? hint : props.hint}
```
Comment (rebelchris) [inline packages/shared/src/components/fields/PasswordField.tsx:64]: But then should this not be the shouldShowStrength comparison?

**Review by rebelchris — APPROVED** [review]
Awesome, just one question for the hint

File: packages/shared/src/components/fields/PasswordField.tsx (hunk 4 above)
Comment (sshanzel) [reply packages/shared/src/components/fields/PasswordField.tsx:64]: I thought there were some ideas behind why this was left out to not use it. I agree it is better to consolidate the conditions using a single parameter. Let me refactor this.

**Pull request merged**